frontend_server/test/server_test.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:args/src/arg_results.dart';
+import 'package:frontend_server/server.dart';
+import 'package:test/test.dart';
+
+class MockedCompiler implements CompilerInterface {
+  var compileCallback;
+  var recompileDeltaCallback;
+  var acceptLastDeltaCallback;
+  var rejectLastDeltaCallback;
+  var invalideCallback;
+
+  MockedCompiler({this.compileCallback, this.recompileDeltaCallback,
+    this.acceptLastDeltaCallback, this.rejectLastDeltaCallback,
+    this.invalideCallback});
+
+  @override
+  compile(String filename, ArgResults options) =>
```

Comment:
alternatively you could define a default callback and directly name it the name of this method:

```dart
class MockCompiler implements CompilerInterface {
   void Function(String, ArgResults) compile;
   void Function(Uri) invalidate;
   void Function()  recompileDelta;
   void Function() acceptLastDelta;

    MockCompiler({this.compile, this.invalidate, ...});
}
```

It is possible that this new Function type syntax is not available everywhere yet, in which case you may want to use a typedef instead.

Yet another alternative: look at `package:mockito`. I'm not sure if it is commonly used in flutter though.